solidity/contracts/isms/hook/AbstractMessageIdAuthorizedIsm.sol

```diff
@@ -113,5 +96,29 @@ abstract contract AbstractMessageIdAuthorizedIsm is
         emit ReceivedMessage(messageId);
     }
 
+    function _statefulVerify(
+        bytes calldata,
+        /*_metadata*/
+        bytes calldata message
+    ) internal returns (bool) {
+        bytes32 messageId = message.id();
+
+        // check for the first bit (used for verification)
+        bool verified = verifiedMessages[messageId].isBitSet(
+            VERIFIED_MASK_INDEX
+        );
+        // rest 255 bits contains the msg.value passed from the hook
+        if (verified) {
+            uint256 _msgValue = verifiedMessages[messageId].clearBit(
+                VERIFIED_MASK_INDEX
+            );
+            if (_msgValue > 0) {
+                verifiedMessages[messageId] -= _msgValue;
+                payable(message.recipientAddress()).sendValue(_msgValue);
```

Comment:
I guess this means that the value can be sent non-atomically before the message arrives

Reply:
imo this is fine but we should clearly document this behavior

Reply:
added